service/controller/app_catalog_entry.go

```diff
@@ -0,0 +1,111 @@
+package controller
+
+import (
+	"github.com/giantswarm/apiextensions/v3/pkg/apis/application/v1alpha1"
+	"github.com/giantswarm/k8sclient/v5/pkg/k8sclient"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"github.com/giantswarm/operatorkit/v4/pkg/controller"
+	"github.com/giantswarm/operatorkit/v4/pkg/resource"
+	"github.com/giantswarm/operatorkit/v4/pkg/resource/wrapper/metricsresource"
+	"github.com/giantswarm/operatorkit/v4/pkg/resource/wrapper/retryresource"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	"github.com/giantswarm/config-controller/pkg/label"
+	"github.com/giantswarm/config-controller/pkg/project"
+	"github.com/giantswarm/config-controller/service/controller/handler/configversion"
+)
+
+type AppCatalogEntryConfig struct {
+	K8sClient k8sclient.Interface
+	Logger    micrologger.Logger
+
+	GitHubToken string
+	UniqueApp   bool
+}
+
+type AppCatalogEntry struct {
+	*controller.Controller
+}
+
+func NewAppCatalogEntry(config AppCatalogEntryConfig) (*AppCatalogEntry, error) {
+	var err error
+
+	resources, err := newAppCatalogEntryResources(config)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	var operatorkitController *controller.Controller
+	{
+		c := controller.Config{
+			K8sClient: config.K8sClient,
+			Logger:    config.Logger,
+			NewRuntimeObjectFunc: func() runtime.Object {
+				return new(v1alpha1.App)
+			},
+			Resources: resources,
+			Selector:  label.AppVersionSelector(config.UniqueApp),
+
+			// Name is used to compute finalizer names. This here results in something
+			// like operatorkit.giantswarm.io/config-controller-app-controller.
```

Comment:
I doubt this comment

Reply:
I took it from here: https://github.com/giantswarm/template-operator/blob/master/service/controller/todo.go#L45-L46
What's wrong with it?

Reply:
oh, i think i read the 'like' here wrong - i'd expect this to end up with `operatorkit.giantswarm.io/config-controller-app-catalog-entry-controller.`

tl;dr i'm a massive nit picker